agent/src/android/hooking.ts

```diff
@@ -63,6 +73,39 @@ export namespace hooking {
     });
   };
 
+
+  export const getClassMethodsOverloads = (className: string, methodsAllowList: string[] = []): Promise<JSON> => {
+    return wrapJavaPerform(() => {
+      const clazz: JavaClass = Java.use(className);
+      const methods = clazz.class.getDeclaredMethods()
+      const result = {}
+      methods.forEach(method => {
+        // This trims out only the function name and uses that to get the overloads
+        const methodName = (method.toGenericString().split('.').filter(part => part.includes('('))[0].split('(')[0])
+        if (methodsAllowList.length === 0 || (methodsAllowList.length > 0 && methodsAllowList.includes(methodName))) {
+          const overloads = clazz[methodName].overloads
+          result[methodName] = {
+            'argTypes': overloads.map(overload => overload.argumentTypes),
+            'returnType': overloads.map(overload => overload.returnType)
+          }
+        }
+      })
+        // Finally append the constructor details
+        try{
+            if (clazz.$init !== undefined){
+                if (methodsAllowList.length === 0 || (methodsAllowList.length > 0 && methodsAllowList.includes("$init"))){
+                  result["$init"] = {
+                    'argTypes': clazz.$init.overloads.map(overload => overload.argumentTypes),  // Return type for constructors are always `void`
+                  }
+                }
+            }
+        }catch {
+            // Some classes don't have constructors, this is a temporary workaround
```

Comment:
What error occurs that we need a try/catch block here? A comment would be useful to know.